packages/ui/Marquee/MarqueeDecorator.js

```diff
@@ -443,6 +443,10 @@ const MarqueeDecorator = hoc(defaultConfig, (config, Wrapped) => {
 			this.distance = null;
 			// Assume the marquee does not fit until calculations show otherwise
 			this.contentFits = false;
+
+			this.setState(state => {
+				return state.overflow === 'ellipsis' ? null : {overflow: 'ellipsis'};
+			});
```

Comment:
I think this makes sense and appears safe enough overall. However, we might need to de-couple this state change bit, since `invalidateMetrics` is called in the `constructor`, and we can't call `setState` there.

Reply:
Good call!

Reply:
Moved that initialization to the constructor rather than calling `invalidateMetrics()`. Adding a new method just to reset the overflow seemed like unnecessary decomposition to me and it's appropriate to set up members in the constructor anyway.